contracts/truefi2/LoanFactory2.sol

```diff
@@ -111,12 +117,24 @@ contract LoanFactory2 is ILoanFactory2, Initializable {
         require(_amount > 0, "LoanFactory: Loans of amount 0, will not be approved");
         require(_term > 0, "LoanFactory: Loans cannot have instantaneous term of repay");
         require(poolFactory.isPool(address(_pool)), "LoanFactory: Pool was not created by PoolFactory");
+        require(loanTokenImplementation != address(0), "Loan token implementation should be set");
 
         uint256 apy = rate(_pool, msg.sender, _amount, _term);
 
         require(apy <= _maxApy, "LoanFactory: Calculated apy is higher than max apy");
 
-        address newToken = address(new LoanToken2(_pool, borrowingMutex, msg.sender, lender, admin, liquidator, _amount, _term, apy));
+        address newToken = Clones.clone(loanTokenImplementation);
```

Comment:
You can just call methods on new LoanToken2(newToken) without doing low-level calls